packages/pds/src/api/com/atproto/identity/signPlcOperation.ts

```diff
@@ -0,0 +1,46 @@
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import * as plc from '@did-plc/lib'
+import { check } from '@atproto/common'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.identity.signPlcOperation({
+    auth: ctx.authVerifier.accessNotAppPassword,
+    handler: async ({ auth, input }) => {
+      const did = auth.credentials.did
+      const { token } = input.body
+      if (!token) {
+        throw new InvalidRequestError(
+          'email confirmation token required to sign PLC operations',
+        )
+      }
+      await ctx.services
+        .account(ctx.db)
+        .assertValidToken(did, 'plc_operation', token)
+
+      const lastOp = await ctx.plcClient.getLastOp(did)
+      if (check.is(lastOp, plc.def.tombstone)) {
+        throw new InvalidRequestError('Did is tombstoned')
+      }
+      const operation = await plc.createUpdateOp(
+        lastOp,
+        ctx.plcRotationKey,
+        (lastOp) => ({
+          ...lastOp,
+          rotationKeys: input.body.rotationKeys ?? lastOp.rotationKeys,
+          alsoKnownAs: input.body.alsoKnownAs ?? lastOp.alsoKnownAs,
+          verificationMethods:
+            input.body.verificationMethods ?? lastOp.verificationMethods,
+          services: input.body.services ?? lastOp.services,
+        }),
+      )
+      return {
+        encoding: 'application/json',
+        body: {
+          operation,
+        },
```

Comment:
Just an idea—we could add a field to this response that indicates whether submitting the op would result in the service no longer being able to submit ops on your behalf.

Reply:
I'm not opposed to it but i don't think it's necessary. If you're interested in having those checks, you can always use `submitPlcOperation`